libraries/chain/block_state.cpp

```diff
@@ -70,6 +71,28 @@ block_state::block_state(const block_state_legacy& bsp) {
    cached_trxs = bsp._cached_trxs;
 }
 
+block_state::block_state(snapshot_detail::snapshot_block_state_v7&& sbs)
+   : block_header_state {
+         .block_id                    = sbs.block_id,
+         .header                      = std::move(sbs.header),
+         .activated_protocol_features = std::move(sbs.activated_protocol_features),
+         .core                        = std::move(sbs.core),
+         .proposal_mtree              = std::move(sbs.proposal_mtree),
+         .finality_mtree              = std::move(sbs.finality_mtree),
```

Comment:
We will not have `proposal_mtree` and `finality_mtree` after #2080. You don't need to add them here or we can remove them after #2080 is resolved.